File: web_version/pmugg_files/network/netGraphMap.js
// This is similar to netMorphism.
ms.netGraphMap = function() {
	this.vertexBtoA = null;
	this.edgeBtoA = null;
};

ms.netGraphMap.create = function(info) {
	var { netB } = info;
	var interiorB = netB.getInterior();

	var map = new ms.netGraphMap();
	map.vertexBtoA = new Array(interiorB.getVertices().length).fill(null);
	map.edgeBtoA   = new Array(interiorB.getEdges()   .length).fill(null);
	return map ;
};

ms.netGraphMap.prototype.copy = function() {
	var result = new ms.netGraphMap();
	result.vertexBtoA = this.vertexBtoA.slice();
	result.edgeBtoA   = this.edgeBtoA.slice();
	return result;
};

File: cpp_version/network/net_graph_map.cpp
#include "pch.h"
#include "net_graph_map.h"
#include "network.h"
// #include "vertex.h"
// #include "edge.h"
// #include "endpoint.h"
// #include "face.h"
#include "../Util/util.h"
#include "net_graph_map_info.h"

namespace ms {

NetGraphMap::NetGraphMap() {}

NetGraphMap* NetGraphMap::copy() const {
    auto* result = new NetGraphMap();
    result->vertexBtoA = vertexBtoA;
    result->edgeBtoA = edgeBtoA;
    return result;
}

NetGraphMap* NetGraphMap::create(const NetGraphMapInfo& info) {
    auto map = new NetGraphMap();
    auto netB = info.networkB;
    map->vertexBtoA.resize(netB->getVertices().size(), nullptr);
    map->edgeBtoA.resize(netB->getEdges().size(), nullptr);
    return map;
}

} // namespace ms 